coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,690 @@
+                # below automatically does that.
+            else:
+                # Prevent people from accidentally defining `process_output`
+                # manually, as this would implicitly override the internally
+                # set-up `process_output`.
+                if hasattr(klass, "process_output"):
+                    raise ValueError("`process_output` is used by given class,"
+                                     " but " + repr(options["output_format"]) +
+                                     " output format was specified.")
+
+                if options["output_format"] == "corrected":
+                    process_output_args = {}
+                    if "diff_severity" in options:
+                        process_output_args["diff_severity"] = (
+                            options["diff_severity"])
+                    if "diff_message" in options:
+                        process_output_args["diff_message"] = (
+                            options["diff_message"])
+
+                    process_output = partialmethod(
+                        process_output_corrected, **process_output_args)
+
+                elif options["output_format"] == "regex":
+                    process_output_args = {
+                        "output_regex": options["output_regex"]}
+                    if "severity_map" in options:
+                        process_output_args["severity_map"] = (
+                            options["severity_map"])
+
+                    process_output = partialmethod(
+                        process_output_regex, **process_output_args)
+
+                else:  # pragma: no cover
+                    # This statement is never reached.
+                    # Due to a bug in coverage we can't use `pass` here, as
+                    # the ignore-pragma doesn't take up this else-clause then.
+                    # https://bitbucket.org/ned/coveragepy/issues/483/partial-
+                    # branch-coverage-pragma-no-cover
+                    assert False
+
+            @classmethod
+            @contextmanager
+            def _create_config(cls, filename, file, **kwargs):
+                """
+                Provides a context-manager that creates the config file if the
+                user provides one and cleans it up when done with linting.
+
+                :param filename:
+                    The filename of the file.
+                :param file:
+                    The file contents.
+                :param kwargs:
+                    Section settings passed from ``run()``.
+                :return:
+                    A context-manager handling the config-file.
+                """
+                content = cls.generate_config(filename, file, **kwargs)
+                if content is None:
+                    yield None
+                else:
+                    tmp_suffix = options["config_suffix"]
+                    with make_temp(suffix=tmp_suffix) as config_file:
```

Comment:
`tmp_suffix` is defined and used only once. Unnecessarily creating a new variable